### PR TITLE
RFC: Add Audit Policies Support

### PR DESCRIPTION
**Pull request by darcyclarke**
[Rendered RFC](https://github.com/npm/rfcs/blob/13d046eaa171f10bb7e8dce4ae0f656adcd33894/accepted/0000-npm-audit-policies.md)

### CODE REVIEW

File: accepted/0000-npm-audit-policies.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+### Implementation
+
+```json
+{
+    "audit": {
+        "policies": [
+            {
+                "name": "Vulnerable",
+                "type": "error",
+                "query": ":vulnerable"
```
Comment (ljharb) [inline accepted/0000-npm-audit-policies.md:28]: the query specifies which packages are selected, but what defines how the results are presented?

Vulnerability output is perhaps a special case, but is the output just "error, nonzero packages showed up in the query!"?

Comment (darcyclarke) [reply accepted/0000-npm-audit-policies.md:28]: Great question - I believe the idea would be to use something like the above but I can put some potential examples of output in the RFC itself

Comment (ljharb) [reply accepted/0000-npm-audit-policies.md:28]: I think that it won't be nearly as useful if all i can do is control the exit code with a query that produces nonzero results. I suspect people will want to provide their own *package* that can be a function that receives the query results as an argument, and can do whatever it likes, as long as it returns an exit code, or something.

**Comment by naugtur** [comment]
Hello!

Where does the JSON go? If it's in a policies file, it feels like I could use it for npm-audit-resolver to specify what to ignore too. 
Can we consider an ignore condition here? Or setting the policy to error on all vulnerable and then specify exceptions with queries for each?

Do at me ;)

**Comment by darcyclarke** [comment]
@naugtur the original idea was to have these policies live in `package.json` & get picked up by `npm audit` (they would then also apply during `install` or `update` if `--audit=true`); I can/should expand the RFC to note this & other specifics left out today (including what inclusion/exclusion might look like - although I think it would be fairly straightforward ie. DSS combinators ex. `:vulnerable:not(.dev)...`, but maybe you're hoping for more flexibility in composition of _which_ policies to run)

**Comment by ljharb** [comment]
I think putting them in `package.json`, thereby inflating the size of the packument and published packages, is probably not the best idea - a separate JSON file seems ideal.

**Comment by ljharb** [comment]
@bnb "too many files" is a subjective opinion - separate files allow you to have granular CODEOWNERS, as well, beyond being data that doesn't belong in a published package. The burden exists regardless because you still have to google for how to configure the package.json field.

**Comment by bnb** [comment]
noting a combination of suggestions from the office hours: if there were a top-level package.json property like `npmConfig`, having the ability to have audit policies _and_ the ability to point to a file (or a module!) rather than having a pre-determined filename, both options could be supported. If policies as a devDep is an option, there's also _so much_ flexibility for both large-scale open source projects and companies alike.

**Comment by ljharb** [comment]
i'd be totally fine with a field that optionally points to a file so that the config is discoverable AND the contents don't have to live in package.json :-)

**Review by rondales — APPROVED** [review]

**Comment by rondales** [comment]
> [Rendered RFC](https://github.com/npm/rfcs/blob/13d046eaa171f10bb7e8dce4ae0f656adcd33894/accepted/0000-npm-audit-policies.md)

**Comment by rondales** [comment]
> > [Rendered RFC](https://github.com/npm/rfcs/blob/13d046eaa171f10bb7e8dce4ae0f656adcd33894/accepted/0000-npm-audit-policies.md)

**Review by rondales — COMMENTED** [review]
accepted/0000-npm-audit-policies.md

**Review by rondales — COMMENTED** [review]
accepted/0000-npm-audit-policies.md

**Pull request closed without merging**